File: typescript/service/reservationService.ts
import Reservation from "../modele/reservation";
import ReservableItem from "../modele/reservableItem";

export default class ReservationService {
  private allReservations: { item: string; reservation: Reservation }[] = [];

  makeReservation(name: string, email: string, item: ReservableItem, start: Date, end: Date): void {
    if (!item.isAvailable(start, end)) {
      throw new Error("Object already reserved");
    }

    const reservation = new Reservation(name, email, start, end);
    item.addReservation(reservation);
    this.allReservations.push({ item: item.name, reservation });
  }

  printReservations(): void {
    if (this.allReservations.length === 0) {
      console.log("No reservation yet");
      return;
    }

    for (const { item, reservation } of this.allReservations) {
      console.log(`📝 ${item} :`);
      reservation.display();
    }
  }
}

File: python/service/reservation_service.py
from modele.reservation import Reservation

class ReservationService:
    def __init__(self):
        self.all_reservations = []

    def make_reservation(self, name, email, item, start_date, end_date):
        if not item.is_available(start_date, end_date):
            raise Exception("Object already reserved")

        reservation = Reservation(name, email, start_date, end_date)
        item.add_reservation(reservation)
        self.all_reservations.append({"item": item.name, "reservation": reservation})

    def print_reservations(self):
        if not self.all_reservations:
            print("No reservation yet")
            return

        for entry in self.all_reservations:
            print(f"📝 {entry['item']} :")
            entry['reservation'].display()
